packages/blade/src/components/Tabs/TabIndicator.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable consistent-return */
+import React from 'react';
+import styled from 'styled-components';
+import { useTabsContext } from './TabsContext';
+import { castWebType, makeMotionTime, useTheme } from '~utils';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+import BaseBox from '~components/Box/BaseBox';
+
+const StyledTabIndicator = styled(BaseBox)(({ theme }) => {
+  return {
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.colors.brand.primary[400],
+    },
+  };
+});
+
+const TabIndicator = ({
+  tabListContainerRef,
+}: {
+  tabListContainerRef: React.RefObject<HTMLElement | null>;
+}): React.ReactElement => {
+  const { theme } = useTheme();
+  const { selectedValue, baseId, variant } = useTabsContext();
+  const [hasMeasured, setHasMeasured] = React.useState(false);
```

Comment:
what does `hasMeasured` mean?

Reply:
It tracks if we are finished with measuring the TabItem's width which we need to calculate to animate the TabIndicator.